lib/validator.rb

```diff
@@ -0,0 +1,22 @@
+class SelectionValidator
+  attr_reader :board
+
+  def initialize(board)
+    @board = board
+  end
+
+  def valid(selection)
+    position = input_to_integer(selection)
+    return position if position.between?(1, 9)
+  end
+
+  def available(position)
+    return position if board.is_available?(position)
+  end
+
+  def input_to_integer(input)
+    Integer(input)
+  rescue
+    0
```

Comment:
What does `0` mean here? Does it have a special meaning? If so, is there some way we can encode that meaning?

Reply:
to_i in Ruby does not throw an error when faced with a non-integer string but instead returns 0.  I did not want to use to_i because it changes a string float to an integer (ex. "1.5" to 1), but I replicated some of the behavior with Integer.